src/frontend/components/DashboardTimedTextPaneConnected/DashboardTimedTextPaneConnected.tsx

```diff
@@ -1,21 +1,46 @@
 import { connect } from 'react-redux';
 import { Dispatch } from 'redux';
 
-import { addMultipleResources } from '../../data/genericReducers/resourceById/actions';
+import { getResourceList } from '../../data/genericReducers/resourceList/actions';
 import { RootState } from '../../data/rootReducer';
+import { ConsumableQuery } from '../../types/api';
 import { appStateSuccess } from '../../types/AppData';
 import { modelName } from '../../types/models';
 import { TimedText } from '../../types/tracks';
 import { DashboardTimedTextPane } from '../DashboardTimedTextPane/DashboardTimedTextPane';
 
-const mapStateToProps = (state: RootState<appStateSuccess>) => ({
+/**
+ * Build props for `<DashboardTimedTextPaneConnected />` from `RootState`.
+ * Intended for internal use, exported for testing purposes only.
+ */
+export const mapStateToProps = (state: RootState<appStateSuccess>) => ({
   jwt: state.context.jwt,
+  timedtexttracks: state.resources[modelName.TIMEDTEXTTRACKS].currentQuery
```

Comment:
Maybe we can factorize this code ? We will need to fetch `timedtexttracks` in other components, at least the `VideoPlayer` one. This block of code is huge, creating a function can be a good enhancement.

Reply:
@lunika we should probably use `reselect`.

If you agree I think I should do this in another PR

Reply:
I never heard about this lib. If you think it's the solution, create a new PR with it :+1: